src/main/java/org/opensearch/timeseries/rest/handler/AbstractTimeSeriesActionHandler.java

```diff
@@ -973,7 +980,9 @@ protected void onSearchConfigNameResponse(SearchResponse response, String name,
 
     protected void tryIndexingConfig(boolean indexingDryRun, ActionListener<T> listener) throws IOException {
         if (!indexingDryRun) {
-            indexConfig(id, listener);
+            boolean isResourceSharingFeatureEnabled = this.settings
```

Comment:
There are a lot of reference to this line of code. It is easy to miss checking backend role filter. Can you put the check to a utility method and call it in similar places?